Sources/TensorFlow/Random.swift

```diff
@@ -18,13 +18,34 @@ import Darwin
 import Glibc
 #endif
 
+/// Generates a new random seed for TensorFlow.
 public func randomSeedForTensorFlow() -> (Int32, Int32) {
     (Int32.random(in: Int32.min..<Int32.max), Int32.random(in: Int32.min..<Int32.max))
 }
 
-//===------------------------------------------------------------------------------------------===//
+//===--------------------------------------------------------------------------------------------------------------===//
 // Random Number Generators
-//===------------------------------------------------------------------------------------------===//
+//===--------------------------------------------------------------------------------------------------------------===//
+
+/// A type-erased random number generator.
+///
+/// The `AnyRandomNumberGenerator` type forwards random number generating operations to an underlying random number 
+/// generator, hiding its specific underlying type.
+public struct AnyRandomNumberGenerator: RandomNumberGenerator {
```

Comment:
I don't see why this type is necessary. `RandomNumberGenerator` can be used as an existential already.

Reply:
We cannot have a stored property in `Context` without using this struct.

Reply:
I addressed the other comments except for this. I don't know if there is another way to define that stored property, but please let me know if you can think of one.

Reply:
Hmm, how so? I think you can add the following to `Context`.

```swift
var randomNumberGenerator: RandomNumberGenerator
```

Reply:
I tried that before but it caused an error. Let me check again and I'll report the error here.

Reply:
So in the standard library calls to e.g. [`Float.random(in:using:)`](https://developer.apple.com/documentation/swift/float/3020582-random) are made using generic types conforming to `RandomNumberGenerator`. If we try to pass something with type `RandomNumberGenerator` we get errors like this: `error: protocol type 'RandomNumberGenerator' cannot conform to 'RandomNumberGenerator' because only concrete types can conform to protocols`. This is why I introduced the type erased version and maintain the generic function calls.

Reply:
Ah I see. Right, protocol existentials don't conform to protocols. [This post](https://forums.swift.org/t/improving-the-ui-of-generics/22814) talked about ways to make this possible in the future, for example:

```swift
extension any RandomNumericGenerator: RandomNumericGenerator {
    ...
}
```

Reply:
Yes I actually found that a bit surprising. Thanks for the reference! I'll take a look at it later today but I imagine that would require more significant changes in the compiler. I can make the new struct internal for now and use it as an implementation details for handling type erasure. Would that be acceptable as a temporary solution?

Reply:
Making the property and the struct internal sounds good!

Reply:
Done. Note that even if this is approved and passes tests, it depends on #319 so we should merge that one first.